code/datums/components/suit_autodoc.dm

```diff
@@ -0,0 +1,345 @@
+#define SUIT_AUTODOC_DAM_MIN 50
+#define SUIT_AUTODOC_DAM_MAX 150
+
+/datum/component/suit_autodoc
+	var/obj/item/healthanalyzer/integrated/analyzer
+
+	var/burn_nextuse
+	var/oxy_nextuse
+	var/brute_nextuse
+	var/tox_nextuse
+	var/pain_nextuse
+
+	var/chem_cooldown = 2.5 MINUTES
+
+	var/enabled = FALSE
+
+	var/damage_threshold = 50
+	var/pain_threshold = 70
+
+	var/list/burn_chems
+	var/list/oxy_chems
+	var/list/brute_chems
+	var/list/tox_chems
+	var/list/pain_chems
+
+	var/datum/action/suit_autodoc_toggle/toggle_action
+	var/datum/action/suit_autodoc_scan/scan_action
+	var/datum/action/suit_autodoc_configure/configure_action
+
+	var/mob/living/carbon/wearer
+
+	var/overdose_threshold_mod = 0.5
+
+/datum/component/suit_autodoc/Initialize(chem_cooldown, list/burn_chems, list/oxy_chems, list/brute_chems, list/tox_chems, list/pain_chems, overdose_threshold_mod)
+	if(!istype(parent, /obj/item))
+		return COMPONENT_INCOMPATIBLE
+	analyzer = new
+	if(!isnull(chem_cooldown))
+		src.chem_cooldown = chem_cooldown
+	if(islist(burn_chems))
+		src.burn_chems = burn_chems
+	else
+		src.burn_chems = list(
```

Comment:
This is not a static list and also this is horrible to look at
```
var/static/list/default_burn_chems = list(
  /datum/reagent/medicine/kelotane,
  /datum/reagent/medicine/tricordrazine,
)
src.burn_chems = burn_chems || default_burn_chems
```
You don't need to sanity check that it's a list because if someone gives something that's not a list it's entirely their own fault.

Reply:
what about isnull to make these the defaults?

Reply:
Why?

Reply:
so you dont need to provide 15 line addcomponent call everytime you want to add this component with the sensible drug lists?

Reply:
No I mean you don't need the isnull check, if you pass in null it will use the default chems if you use the code I put above